src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java

```diff
@@ -543,15 +543,20 @@ private void validateScheduledEventsRequest(final LocalDate fromDate, final Loca
      */
     public void verifyBookingAccess(final Long bookingId, boolean accessDeniedError, final String... rolesAllowed) {
 
-        checkBookingExists(bookingId);
-        if (hasAnySystemOverrideRole(rolesAllowed)) return;
+        Objects.requireNonNull(bookingId, "bookingId is a required parameter");
+
+        if (hasAnySystemOverrideRole(rolesAllowed)) {
+            checkBookingExists(bookingId);
+            return;
+        }
 
         final var agencyIds = agencyService.getAgencyIds(false);
         if (AuthenticationFacade.hasRoles("INACTIVE_BOOKINGS")) {
             agencyIds.addAll(Set.of("OUT", "TRN"));
         }
 
         if (agencyIds.isEmpty()) {
+            checkBookingExists(bookingId);
```

Comment:
All of the "if" blocks are starting to look a bit unreadable.  This should hopefully start to tidy up once we can remove the checks for different types of exceptions to throw and also we will not need to do explicit client only checks